src/options.c

```diff
@@ -533,24 +561,28 @@ char *optionsNameThumbnail(const char *name)
 void optionsParseAutoselect(char *optarg)
 {
     char *token;
-    const char tokenDelimiter[2] = ",";
-    int dimensions[4];
+    int *dimensions[] = {&opt.autoselectX, &opt.autoselectY, &opt.autoselectW,
+        &opt.autoselectH, NULL /* Sentinel. */};
     int i = 0;
+    const char *errmsg;
+
+    /* Geometry dimensions must be in format x,y,w,h */
+    token = strtok(optarg, ",");
+    for (; token != NULL; token = strtok(NULL, ",")) {
+        if (dimensions[i] == NULL)
+            errx(EXIT_FAILURE, "option --autoselect: too many dimensions");
+
+        *dimensions[i] = options_parsenum(token, 1, INT_MAX, &errmsg);
```

Comment:
On the `master` branch you can use `autoselect` from range 0: `scrot -a 0,0,10,10`
The minor range must start at 0.

Making the change in both branches gives the same result.
```bash
382ee633c173f1ed1f44f61b7cf54a8e  2023-01-06-182253_10x10_scrot.png
382ee633c173f1ed1f44f61b7cf54a8e  2023-01-06-182319_10x10_scrot.png <-- master